ui/client/components/modals/CalculateCountsDialog.tsx

```diff
@@ -0,0 +1,150 @@
+import moment, {Moment, MomentInput} from "moment"
+import React, {useCallback, useState} from "react"
+import DateTimePicker from "react-datetime"
+import {useDispatch, useSelector} from "react-redux"
+import {fetchAndDisplayProcessCounts} from "../../actions/nk"
+import {getProcessId} from "../../reducers/selectors/graph"
+import "../../stylesheets/visualization.styl"
+import {ButtonWithFocus} from "../withFocus"
+import Dialogs from "./Dialogs"
+import GenericModalDialog from "./GenericModalDialog"
+
+const datePickerStyle = {
+  className: "node-input",
+}
+
+const dateFormat = "YYYY-MM-DD"
+const timeFormat = "HH:mm:ss"
+
+interface Range {
+  name: string,
+  from: () => Moment,
+  to: () => Moment,
+}
+
+const predefinedRanges: Range[] = [
+  {
+    name: "Last hour",
+    from: () => moment().subtract(1, "hours"),
+    to: () => moment(),
+  },
+  {
+    name: "Today",
+    from: () => moment().startOf("day"),
+    to: () => moment(),
+  },
+  {
+    name: "Yesterday",
+    from: () => moment().subtract(1, "days").startOf("day"),
+    to: () => moment().startOf("day"),
+  },
+  {
+    name: "Day before yesterday",
+    from: () => moment().subtract(2, "days").startOf("day"),
+    to: () => moment().subtract(1, "days").startOf("day"),
+  },
+  {
+    name: "This day last week",
+    from: () => moment().subtract(8, "days").startOf("day"),
+    to: () => moment().subtract(7, "days").startOf("day"),
+  },
+]
+
+type PickerProps = {label: string, onChange: (date: MomentInput) => void, value: Date}
+const Picker = ({label, onChange, value}: PickerProps): JSX.Element => (
+  <>
+    <p>{label}</p>
+    <div className="datePickerContainer">
+      <DateTimePicker
+        dateFormat={dateFormat}
+        timeFormat={timeFormat}
+        inputProps={datePickerStyle}
+        onChange={onChange}
+        value={value}
+      />
+    </div>
+  </>
+)
+
+type RangeButtonProps = {range: Range, onChange: (value: [Moment, Moment]) => void}
+const RangeButton = ({range, onChange}: RangeButtonProps): JSX.Element => {
+  const {from, to, name} = range
+  const onClick = useCallback(
+    () => onChange([from(), to()]),
+    [from, to],
+  )
+  return (
+    <ButtonWithFocus type="button" title={name} className="predefinedRangeButton" onClick={onClick}>
```

Comment:
Can we also refactor `predefinedRangeButton` to `predefined-range-button`? After that we will have more styles in one standard.

Reply:
I will not touch the styles topic now. next pr will bring visual changes